flink-connector-aws/flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/sink/DynamoDbBeanElementConverter.java

```diff
@@ -54,18 +55,17 @@ public DynamoDbBeanElementConverter(final Class<InputT> recordType, final boolea
 
     @Override
     public DynamoDbWriteRequest apply(InputT element, SinkWriter.Context context) {
-        if (tableSchema == null) {
-            // We have to lazily initialize this because BeanTableSchema is not serializable and
-            // there is no open() method on ElementConverter (FLINK-29938)
-            tableSchema = createTableSchema(recordType);
```

Comment:
nit: Can we add a `Preconditions.checkNotNull(...)` here with a nice error message instead?